### DISCUSSION TITLE
Reconnect to LDAP via load balancer hostname

### DISCUSSION THREAD

**Discussion by davelee53** [rabbitmq/rabbitmq-server#3089] [Other]
hi,
   We are running into the issue that our RabbitMQ LDAP plugin is failing to reconnect to the LDAP cluster via a loadbalancer.  We have some ldap server nodes that are all connected via the loadbalancer hostname like ldaps://ldap.  
We periodically have to cycle our ldap server nodes for maintenance.  Therefore any cached/pooled connections to the ldap node that is being cycled will be disconnected. It seems this ldap plugin is not able to purge those connections if they were established via virtual or load balancer hostname.   We are currently using RabbitMQ v3.7.25 which should have the fix from this [previous reported issue.](https://github.com/rabbitmq/rabbitmq-auth-backend-ldap/issues/65#issuecomment-295654745)
Our current configurations:
auth_ldap.servers.1 = ldap   (a single connection to the loadbalancer host)
auth_ldap.connection_pool_size = 256
auth_ldap.idle_timeout = 50000  

Thanks!

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
The LDAP client used by RabbitMQ does not know whether it's connecting to a load balancer or not. When upstreams are "cycled" (restarted?), it's up to your load balancer to close the client side of the connection. Otherwise the client would not know that's been disconnected.

LDAP connection pooling in this client works the same way regardless of whether a load balancer is used: it will purge only those connections that have been closed (by the load balancer or an actual LDAP server node).

**Reply by davelee53** [reply]
Thanks for the reply.
 "...Otherwise the client would not know that's been disconnected."  I am curious about this.  If a connection has been disconnected from within a ldap server node, are you saying the client, the ldap plugin, would not able to detect it?
When a connection is first established via a virtual host/loadbalancer, doesn't it cache that connection to the actual physical ldap node? If it does, I would hope the plugin will be notified when the connection is dropped and does its purging and removes it from the pool. Our loadbalancer does nothing but routing the request to one of its ldap cluster nodes.
